Rakefile

```diff
@@ -46,10 +48,29 @@ if !Puma.jruby?
     CLEAN.include "lib/puma/puma_http11.rb"
   end
 else
-  # Java (JRuby)
+  task :compile => :maven_dependencies
+  # create a task which will download the dependencies if they do not exist already in the working directory
+  task :maven_dependencies do
+    %w(https://repo1.maven.org/maven2/io/netty/netty-common/4.1.47.Final/netty-common-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-codec/4.1.47.Final/netty-codec-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-buffer/4.1.47.Final/netty-buffer-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-handler/4.1.47.Final/netty-handler-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.29.Final/netty-tcnative-boringssl-static-2.0.29.Final.jar).each do |dependency|
+      filename = URI(dependency).path.split('/').last
+      next if File.exists?(filename)
+      response = Net::HTTP.get_response(URI(dependency))
+      open(filename, "wb") { |file|
```

Comment:
It would be better if we wrote this to the `pkg` directory (which you'll have to make if it doesnt exist ofc).

Reply:
wrote it to the _tmp/java/puma_http11_ directory